### PR TITLE
[android][test-suite] upgrade JSC to 224109 revision

### PR DESCRIPTION
**Pull request by esamelson**
# Why

port of https://github.com/expo/universe/pull/2908 . Depends on https://github.com/expo/expo/pull/2367

# How

Having this configuration in expoview only doesn't seem to work, so I've added `jsc-android` as a dependency of the expokit npm package.

# Test Plan

Testing NCL & test-suite and also just loading an app a bunch of times.

CI is failing because #2367 needs to be merged before this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "workspaces": {
     "packages": [
       "apps/*",
+      "expokit-npm-package",
```
Comment (ide) [inline package.json]: Could this package live under "packages/expokit"? (If so, would like to move it in another PR)

**Review by ide — APPROVED** [review]
Looks good. Let's get the PR to drop Android 4.4 / API 19 in and then land this with ample time to test more.

File: package.json (hunk 1 above)
Comment (esamelson) [reply package.json]: yeah, I was thinking that as well. Will just take a bit of work to migrate (I think there are still scripts in universe that depend on this location).
